pandas/core/dtypes/missing.py

```diff
@@ -698,6 +698,12 @@ def is_valid_na_for_dtype(obj, dtype: DtypeObj) -> bool:
     """
     if not lib.is_scalar(obj) or not isna(obj):
         return False
+    elif lib.is_pyarrow_scalar(obj):
+        return (
+            obj.is_null()
+            and hasattr(dtype, "pyarrow_dtype")
```

Comment:
is this a proxy for `isinstance(dtype, ArrowDtype)`?

Reply:
xref #51378

Reply:
> is this a proxy for `isinstance(dtype, ArrowDtype)`?

Yes, I wanted to ask about that: currently this files does not import anything from pandas.core.arrays (only from libs and other core.dtypes files). I imagine that is on purpose? 
It's a bit a strange situation with some of our own dtypes being defined in core.dtypes.dtypes (and so those can be imported), and some others in core.arrays

Reply:
> I imagine that is on purpose?

Yes.  The idea was that core.dtypes can be mostly before/above the rest of core in the dependency structure (which is reflected in the isort order).  If importing ArrowDtype here avoids a code smell, I won't object.

> It's a bit a strange situation with some of our own dtypes being defined in core.dtypes.dtypes (and so those can be imported), and some others in core.arrays

Agreed.  I've been thinking recently it would be nice to move ArrowDtype and SparseDtype to core.dtypes.dtypes.